src/main/java/com/elixr/poc/rest/response/PurchaseGetResponse.java

```diff
@@ -0,0 +1,22 @@
+package com.elixr.poc.rest.response;
+
+import com.elixr.poc.data.Purchase;
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.List;
+
+/**
+ * Send successful response to user while reterival operations.
+ */
+@Data
+public class PurchaseGetResponse extends ErrorResponse {
+
+    private List<Purchase> purchase;
+
+    @Builder
+    public PurchaseGetResponse(boolean success, List<Purchase> purchase) {
+        setSuccess(success);
```

Comment:
If you are using the success attribute to indicate status of a request, then I would suggest you to rename the abstract class to something like Response or AppResponse etc.

Reply:
Updated